web/analytics/models.py

```
@@ -0,0 +1,109 @@
+    def completion_percentage(self):
+        progress = CourseProgress.objects.filter(
+            enrollment__student=self.student,
+            enrollment__course=self.course
+        ).first()
+        
+        if progress:
+            return progress.completion_percentage
+        return 0
+    
+    @property
+    def engagement_score(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `engagement_score`

(ANN201)